docs/code-standards.md

```diff
@@ -0,0 +1,35 @@
+# Code & Documentation Standards
+
+This document details our goals for code quality, coverage and documentation standards. At the time
+of this writing, these are standards that all of our code may not meet. We should always try to 
+improve our code to meet them.
+
+## Code Quality And Documentation Standards
+
+We aim to write clear code that is easy to read and comprehend. In particular, that goal means the
+following:
+
+1. All Go code must pass the Go linter checks
+    - We have these checks integrated into our CI system
+2. All Go code should pass the `go vet`
+    - These checks are not integrated into our CI system
+    - Anyone with the Go toolchain installed on their system can run the vet checks by executing the
+    following command on their machine:
+
+    ```go
+    go vet ./pkg/... ./cmd/... ./test/...
+    ```
+3. All Go code must be formatted with [gofmt](https://golang.org/cmd/gofmt/)
+    - Anyone can run the formatter by runinng `make format` on their machine
+4. Any exported symbols (`type`s, `interface`s, `func`s, `struct`s, etc...) must have Godoc
```

Comment:
Does it matter whether it comes from `gofmt`, `go vet`, or `golint`? This is widely held as a best practice and I know @arschles and myself are both sticklers about this. Even if no tool complained about this, I'd _still_ want to mention it.